core/frontend/src/components/autopilot/AutopilotManagerUpdater.vue

```diff
@@ -76,6 +79,22 @@ export default Vue.extend({
           notifications.pushError({ service: autopilot_service, type: 'AUTOPILOT_BOARD_FETCH_FAIL', message })
         })
     },
+    async fetchFirmwareInfo(): Promise<void> {
+      try {
+        const response: AxiosResponse = await back_axios({
+          method: 'get',
+          url: `${autopilot.API_URL}/firmware_info`,
+          timeout: 10000,
+        })
+        const firmware_info: FirmwareInfo = response.data
+        autopilot.setFirmwareInfo(firmware_info)
+      } catch (error) {
+        autopilot.setFirmwareInfo(null)
+        if (error === backend_offline_error) { return }
+        const message = error.response.data.detail ?? error.message
+        notifications.pushError({ service: autopilot_service, type: 'AUTOPILOT_FIRM_INFO_FETCH_FAIL', message })
+      }
```

Comment:
I prefer to keep the await approach. For me, it's more clear when both writhing and reading, specially because it becomes explicit that a promise is being resolved.